addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py

```diff
@@ -701,6 +703,13 @@ def _get_document_type_code_vals(self, invoice, invoice_data):
     # IMPORT
     # -------------------------------------------------------------------------
 
+    def _import_invoice_ubl_cii(self, invoice, file_data, new=False):
+        """ Allow AttachedDocument files to have their original record properly parsed. """
+        if file_data['xml_tree'].tag.split('}')[-1] == 'AttachedDocument':
+            file_data['xml_tree'] = self._ubl_parse_attached_document(file_data['xml_tree'])
+
+        return super()._import_invoice_ubl_cii(invoice, file_data, new=new)
```

Comment:
I would put this logic in the `_get_edi_decoder` method in `account_move.py`, and check there whether the document is an AttachedDocument, and if it is, extract the embedded attachment *before* we call `_get_ubl_cii_builder_from_xml_tree`. That way, you can make sure that `_get_ubl_cii_builder_from_xml_tree` looks at the *inner document* to identify its type.